sqlmesh/core/model/meta.py

```diff
@@ -284,6 +327,14 @@ def _refs_validator(cls, vs: t.Any, info: ValidationInfo) -> t.List[exp.Expressi
     def ignored_rules_validator(cls, vs: t.Any) -> t.Any:
         return LinterConfig._validate_rules(vs)
 
+    @field_validator("grants_target_layer", mode="before")
+    def _grants_target_layer_validator(cls, v: t.Any) -> t.Any:
+        return cls._validate_str_enum_value(v)
+
+    @field_validator("virtual_environment_mode", mode="before")
+    def _virtual_environment_mode_validator(cls, v: t.Any) -> t.Any:
```

Comment:
The 2 fields can share the same validator. There's no point in having 2 and a separate `_validate_str_enum_value` class method